pkg/trojan/message.go

```diff
@@ -89,14 +89,33 @@ func NewMessage(topic Topic, payload []byte) (Message, error) {
 // Wrap creates a new trojan chunk for the given targets and Message
 // a trojan chunk is a content-addressed chunk made up of span, a nonce, and a payload which contains the Message
 // the chunk address will have one of the targets as its prefix and thus will be forwarded to the neighbourhood of the recipient overlay address the target is derived from
-func (m *Message) Wrap(targets Targets) (swarm.Chunk, error) {
+// this is done by iteratively enumerating different nonces until the BMT hash of the serialization of the trojan chunk fields results in a chunk address that has one of the targets as its prefix
+func (m *Message) Wrap(ctx context.Context, targets Targets) (swarm.Chunk, error) {
```

Comment:
out of scope for this PR, but I don't understand why we need `Targets` as a type alias for `[]Target`. we should probably get rid of this as this complicates things in other parts of the codebase